neuraxle/checkpoints.py

```diff
@@ -19,185 +19,558 @@
+
+        :param data_container: data container to read checkpoint for
+        :type data_container: DataContainer
+        :param context: execution context to read checkpoint from
+        :type context: ExecutionContext
+        :return: data container checkpoint
+        :rtype: DataContainer
         """
+        data_inputs = []
+        for current_id in data_container.current_ids:
+            checkpoint = self.data_input_checkpointer.read(
+                current_id=current_id,
+                checkpoint_path=context.get_path()
+            )
+            data_inputs.append(checkpoint)
+        data_container.set_data_inputs(data_inputs)
+
+        if self.expected_output_checkpointer is None:
+            data_container.set_expected_outputs(None)
+            return data_container
+
+        expected_outputs = []
         for current_id in data_container.current_ids:
-            if not os.path.exists(self.get_checkpoint_file_path(current_id)):
+            checkpoint = self.expected_output_checkpointer.read(checkpoint_path=context.get_path(), current_id=current_id)
+            expected_outputs.append(checkpoint)
+        data_container.set_expected_outputs(expected_outputs)
+
+        return data_container
+
+    def should_resume(self, data_container: DataContainer, context: ExecutionContext) -> bool:
+        """
+        Returns if the whole data container has been checkpointed.
+
+        :param data_container: data container to read checkpoint for
+        :type data_container: DataContainer
+        :param context: execution context to read checkpoint from
+        :type context: ExecutionContext
+        :return: data container checkpoint
+        :rtype: DataContainer
+        """
+        for current_id, data_input, _ in data_container:
+            if not self.data_input_checkpointer.checkpoint_exists(checkpoint_path=context.get_path(), current_id=current_id):
+                return False
+
+        if self.expected_output_checkpointer is None:
+            if context.get_execution_mode() == ExecutionMode.FIT or context.get_execution_mode() == ExecutionMode.FIT_TRANSFORM:
+                warnings.warn('Resuming Data Checkpoint {0} without an expected output checkpointer with Step Saver {1}. Perhaps, you wanted to have an expected output checkpointer ?'.format(context.get_path(), self.__class__.__name__))
+            return True
+
+        for current_id, _, expected_output in data_container:
+            if not self.expected_output_checkpointer.checkpoint_exists(checkpoint_path=context.get_path(), current_id=current_id):
                 return False
 
         return True
 
-    def get_checkpoint_file_path(self, current_id) -> str:
-        """
-        Returns the checkpoint file path for a data input id
 
-        :param current_id:
-        :return:
-        """
-        return os.path.join(
-            self.checkpoint_path,
-            '{0}.pickle'.format(current_id)
+class MiniCheckpoint(Checkpoint):
```

Comment:
This is not a `MiniCheckpoint`: rename class. 

It is in fact a kind of full checkpoint, but it make use of mini checkpointers and a wrapper inside itself.

Reply:
This confused me a lot when reading the test because the checkpoint is full.

Reply:
Renamed to DefaultCheckpoint. couldn't find a better name